### PR TITLE
Fix / Chat wiggle issues

### PR DESCRIPTION
**Pull request by Bratanov**
# Problem

This PR tries to solve the chat scroll position issues specified in: #38 and more precisely the:

1)

> When song name is at the edge of the container it wiggles because reflowing changes the scroll height. Changing scroll height on the other hand triggers returning of the song to it's original place. The end result is flikering and it's bloody annoying.

2)

> when song name is at the bottom of chat container it can reflow to the top but it cannot return back since it's original position is no longer available (because of the scroll height change)

and possibly (couldn't reproduce if this is fixed, but it might very well be):

10)

> Bug with currently playing song in chat:

# Changes

- Changed the sticky message to be a separate element, so the original element can stay in chat and hold onto it's place. Not hiding the original element will visually not be a problem, since the element should be hidden from view when the sticky is visible. (we can still barely see it, but who cares)

- We now update both the sticky and the non-sticky with the currently playing song info. The sticky is updated with the new info when a new_song event comes in.

- Moved the whole `renderer` object higher up in the file, because as I started working on it it somehow became unavailable to some of the methods higher up

# Remaining issues

- The first element/message in chat will be hidden when the sticky is visible. I couldn't CSS the things correctly, so I left it as it is. It's not that big of an issue, feel free to suggest improvements for it. A downside is that the scrollbars will also be partially hidden when the sticky is shown, feel free to suggest improvements for this one too.

### CODE REVIEW

**Comment by IvayloAtanasov** [comment]
Pretty cool solution. Both issues you've listed annoy me too, but I guess it's the "less evil" between:
1. having overlapped scroll and first message
2. having wiggling messages

I've also tried to come up with something creative here, but with no luck. Maybe if we try to use some kind of custom div with on drag event handler, instead of the native scroll we can achieve what we want, but this sounds a bit complex/potentially breaking to me. Sooo feel free to merge whenever you like.

**Review by IvayloAtanasov — APPROVED** [review]
:turkey:

**Comment by Bratanov** [comment]
Added a few improvements. Looks better with a one-line sticky, than with multi-line sticky, but it's something. I think that's all I want to add here for now, so I'll merge if no other issues.
image

File: public/sass/components/components.chat.scss [hunk 1]
```diff
@@ -32,6 +32,15 @@
...
     overflow-y: scroll;
     overflow-x: hidden;
+
+    &--with-sticky::-webkit-scrollbar-track {
+      margin-top: 2rem;
+    }
+  }
+
+  &__start {
+    margin: 100px auto;
```
Comment (IvayloAtanasov) [inline public/sass/components/components.chat.scss:42]: I would have used flex here to do the centering but let's say this is OK as well :see_no_evil:

**Comment by IvayloAtanasov** [comment]
Pretty cool!
Another appropriate chat related song that comes on the top of my head is [Chat chat Lenche](https://textove.com/kondyo-chat-chat-lenche-tekst). I have some doubts that it's a bit too exquisite for mainstream audience, though...

File: public/sass/components/components.chat.scss (hunk 1 above)
Comment (Bratanov) [reply public/sass/components/components.chat.scss:42]: Let's not put the horse in front of the karutza. We can improve if needed 😅

**Pull request merged**